code/main.cpp

```diff
@@ -1,27 +1,39 @@
-#include <hwlib.hpp>
 #include <comm.hpp>
+#include <hwlib.hpp>
 
-#include <i2c_bus.hpp>
+#include <display_module.hpp>
+#include <st7735_buffered.hpp>
+#include <display_screen.hpp>
 #include <ssd1306_oled_buffered.hpp>
 
-#include "display_module.hpp"
-
 int main() {
     // kill the watchdog
     WDT->WDT_MR = WDT_MR_WDDIS;
 
     hwlib::wait_ms(10);
+    
+    // hardware spi
+    auto spi_bus = due::hwspi(0, due::hwspi::SPI_MODE::SPI_MODE0, 3);
+    auto pin_dummy = hwlib::pin_out_dummy;
+    auto dc = hwlib::target::pin_out(hwlib::target::pins::d9);
+    auto rst = hwlib::target::pin_out(hwlib::target::pins::d8);
 
-    // Initialise a I2C bus at full speed (400Kbit/s)
-    auto bus = r2d2::i2c::i2c_bus_c(
-        r2d2::i2c::i2c_bus_c::interface::interface_0, 400000);
 
-    // Initialise the display using address 0x3C and the bus
-    auto display = r2d2::display::ssd1306_oled_buffered_c(bus, 0x3c);
+    // use pin_dummy because chip select(cs) is controlled by the hardware spi
+    r2d2::display::st7735_buffered_c<
+        static_cast<std::size_t>(r2d2::claimed_display_cursor::CURSORS_COUNT),
```

Comment:
Why does it need a cursor count template parameter? This value is available is available in the class as well right?

Reply:
the cursorcount is variable since you dont know how many cursors wil be used before connection with internal communication. which is astablished after construction

Reply:
You can include `frame_enums.hpp` and get the `claimed_display_cursor::CURSORS_COUNT` directly right?

Reply:
fixed in https://github.com/R2D2-2019/display/pull/36/commits/7a0d3f0bcd347f0369be52a115a961d473c0e5ce